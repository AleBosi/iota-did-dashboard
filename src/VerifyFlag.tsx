import React, { useEffect, useState } from "react";
import { verifyProof } from "./utilsVC";
import { Loader2 } from "lucide-react";

export default function VerifyFlag({ vc }: { vc: any }) {
  const [valid, setValid] = useState<null | boolean>(null);

  useEffect(() => {
    let mounted = true;
    verifyProof(vc).then(res => {
      if (mounted) setValid(res);
    });
    return () => {
      mounted = false;
    };
  }, [vc]);

  return (
    <span className="text-xl ml-2">
      {valid === null ? (
        <Loader2
          className="animate-spin text-gray-500 w-5 h-5 inline-block"
          title="Verifica in corso"
        />
      ) : valid ? (
        <span className="text-green-600" title="Firma valida">✅</span>
      ) : (
        <span className="text-red-500" title="Firma non valida">❌</span>
      )}
    </span>
  );
}
