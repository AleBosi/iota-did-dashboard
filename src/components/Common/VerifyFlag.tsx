import { useEffect, useState } from "react";
import { verifyVc } from "../../services/vc";

export default function VerifyFlag({ vc }: { vc: any }) {
  const [ok, setOk] = useState<boolean | null>(null);
  useEffect(() => {
    (async () => setOk(await verifyVc(vc)))();
  }, [vc]);
  if (ok === null) return <span className="text-gray-500">…</span>;
  return ok ? <span className="text-green-600">✅</span> : <span className="text-red-600">❌</span>;
}
