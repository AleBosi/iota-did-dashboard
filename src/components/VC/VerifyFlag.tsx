import React, { useEffect, useState } from "react";
import type { VerifiableCredential } from "../../models/vc";
import { verifyVC } from "../../utils/vcIntegrity";

interface Props {
  vc: VerifiableCredential;
}

/**
 * VerifyFlag.tsx
 * UI-only component:
 * - Verifica integrità di una VC tramite verifyVC (fonte unica).
 * - Mostra ✅ se valida, ❌ se non valida (con tooltip motivazione).
 */
export default function VerifyFlag({ vc }: Props) {
  const [result, setResult] = useState<{ ok: boolean; reason?: string }>({ ok: false });

  useEffect(() => {
    let active = true;
    (async () => {
      const res = await verifyVC(vc);
      if (active) {
        setResult({ ok: res.valid, reason: res.reason });
      }
    })();
    return () => {
      active = false;
    };
  }, [vc]);

  return (
    <span
      className={`ml-2 text-xl ${result.ok ? "text-green-600" : "text-red-600"}`}
      title={result.ok ? "VC integra" : result.reason || "VC non valida"}
    >
      {result.ok ? "✅" : "❌"}
    </span>
  );
}
