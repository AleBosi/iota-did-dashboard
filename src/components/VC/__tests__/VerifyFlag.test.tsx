import { render, screen } from "@testing-library/react";
import VerifyFlag from "../VerifyFlag";

describe("VerifyFlag component", () => {
  it("mostra la spunta verde ✅ e la classe text-green-600 quando isValid è true", () => {
    render(<VerifyFlag isValid={true} />);
    const flag = screen.getByText("✅");
    expect(flag).toBeInTheDocument();
    expect(flag).toHaveClass("text-green-600");
  });

  it("mostra la X rossa ❌ e la classe text-red-600 quando isValid è false", () => {
    render(<VerifyFlag isValid={false} />);
    const flag = screen.getByText("❌");
    expect(flag).toBeInTheDocument();
    expect(flag).toHaveClass("text-red-600");
  });

  it("matcha lo snapshot", () => {
    const { asFragment } = render(<VerifyFlag isValid={true} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
