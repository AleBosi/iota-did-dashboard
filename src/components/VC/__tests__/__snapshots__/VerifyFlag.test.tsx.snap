// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`VerifyFlag component matcha lo snapshot 1`] = `
<DocumentFragment>
  <span
    class="ml-2 text-xl text-green-600"
  >
    âœ…
  </span>
</DocumentFragment>
`;
